app/src/main/java/com/example/next_show/adapters/ShowAdapter.java

```diff
@@ -54,7 +52,11 @@ public void clear() {
     }
 
     public void addAll(List<Show> list) {
-        shows.addAll(list);
+        List<Show> modifiedShows = list;
+        if(fromSavedFragment){
+            modifiedShows = Show.fromParseShows(list);
+        }
+        shows.addAll(modifiedShows);
         notifyDataSetChanged();
```

Comment:
Hm, we should probably try to see if we can, instead of having a boolean, pass in this logic based on where this is used (like callbacks that we talked about) instead. 

Boolean flags often aren't great for a variety of reasons, but they also don't scale.  So, what you often want to do is just pass in some interface that will run the right code based on where the object is created.  Then this adapter doesn't need to know any information about where it was created -- it can just consume the same API either way